registry.go

```diff
@@ -0,0 +1,220 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// RegistryService handles communication with the container registry related methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryService struct {
+	client *Client
+}
+
+// RegistryRepository represents a GitLab content registry repository.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryRepository struct {
+	ID        int        `json:"id"`
+	Name      string     `json:"name"`
+	Path      string     `json:"path"`
+	Location  string     `json:"location"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+}
+
+func (s RegistryRepository) String() string {
+	return Stringify(s)
+}
+
+// RegistryRepositoryTag represents a GitLab registry image tag.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryRepositoryTag struct {
+	Name     string `json:"name"`
+	Path     string `json:"path"`
+	Location string `json:"location"`
+}
+
+func (s RegistryRepositoryTag) String() string {
+	return Stringify(s)
+}
+
+// RegistryRepositoryTagDetail represents a GitLab registry tag detail.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryRepositoryTagDetail struct {
+	Name          string     `json:"name"`
+	Path          string     `json:"path"`
+	Location      string     `json:"location"`
+	Revision      string     `json:"revision"`
+	ShortRevision string     `json:"short_revision"`
+	Digest        string     `json:"digest"`
+	CreatedAt     *time.Time `json:"created_at,omitempty"`
+	TotalSize     int        `json:"total_size"`
+}
+
+func (s RegistryRepositoryTagDetail) String() string {
+	return Stringify(s)
+}
+
+// ListRegistryRepositoriesOptions represents the available ListRegistryRepositories() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/container_registry.html#list-registry-repositories
+type ListRegistryRepositoriesOptions ListOptions
+
+// ListRegistryRepositoryTagsOptions represents the available ListRegistryRepositoryTags() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/container_registry.html#list-repository-tags
+type ListRegistryRepositoryTagsOptions ListOptions
+
+// BulkDeleteRegistryRepositoryTagsOptions represents the available BulkDeleteRegistryRepositoryTags() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/container_registry.html#delete-repository-tags-in-bulk
+type BulkDeleteRegistryRepositoryTagsOptions struct {
```

Comment:
I would call this `DeleteRegistryRepositoryTagsOptions` to make it just a bit shorter. I understand that the methods names would them be `DeleteRegistryRepositoryTag` vs `DeleteRegistryRepositoryTags` which is very similar, but still clear enough I think.